File: lib/languages/python/if-else.py
# Basic if/else
def check_number(x):
   if x > 0:
       return "Positive"
   elif x < 0:
       return "Negative"
   else:
       return "Zero"

# Conditional expression (ternary operator)
is_admin = True  # Example variable
message = "Welcome, admin!" if is_admin else "Welcome, user!"

# If without else
def maybe_log(message, verbose):
   if verbose:
       print(message)

# Nested conditions
def process_data(data):
   if data is not None:
       if len(data) > 0:
           return f"Valid data: {data}"
       else:
           return "Empty data"
   else:
       return "No data"

# Logical operators in conditions
def validate_user(user):
   if user and hasattr(user, 'is_active') and user.is_active and user.age >= 18:
       return True
   return False

# Checking multiple conditions
def classify_temperature(temp):
   if temp < 0:
       return "Freezing"
   elif temp < 10:
       return "Cold"
   elif temp < 20:
       return "Cool"
   elif temp < 30:
       return "Warm"
   else:
       return "Hot"

# Truthy and falsy values
def process_value(value):
   if value:  # Checks if value is truthy
       return "Value is truthy"
   else:
       return "Value is falsy"

# Falsy values: None, False, 0, "", [], {}, set()
# Everything else is truthy

File: lib/languages/kotlin/if-else.kt
// Basic if/else
fun checkNumber(x: Int): String {
   if (x > 0) {
       return "Positive"
   } else if (x < 0) {
       return "Negative"
   } else {
       return "Zero"
   }
}

// Ternary operator (expression form of if/else)
val isAdmin = true // Example variable
val message = if (isAdmin) {
   "Welcome, admin!"
} else {
   "Welcome, user!"
}

// If with multiple statements in blocks
val x = 5 // Example variable
val result = if (x > 0) {
   println("Positive number")
   "Positive"
} else {
   println("Non-positive number")
   "Non-positive"
}

// If with no else (returns Unit implicitly)
fun maybeLog(message: String, verbose: Boolean) {
   if (verbose) {
       println(message)
   }
}

// Nested conditions
fun processData(data: String?): String {
   if (data != null) {
       if (data.length > 0) {
           return "Valid data: $data"
       } else {
           return "Empty data"
       }
   } else {
       return "No data"
   }
}

// Logical operators in conditions
fun validateUser(user: User?): Boolean {
   if (user != null && user.isActive && user.age >= 18) {
       return true
   }
   return false
}

// Using when as an alternative to if/else chains
fun classifyTemperature(temp: Int): String {
   return when {
       temp < 0 -> "Freezing"
       temp < 10 -> "Cold"
       temp < 20 -> "Cool"
       temp < 30 -> "Warm"
       else -> "Hot"
   }
}

// Elvis operator for null checks
fun getLength(text: String?): Int {
   return text?.length ?: 0
}

// Safe calls with let
fun processNonEmptyString(text: String?): String {
   return text?.let {
       if (it.isNotEmpty()) {
           "Processing: $it"
       } else {
           "Empty string"
       }
   } ?: "Null string"
}

// Example User class
class User(val name: String, val age: Int, val isActive: Boolean = true)

